src/main/java/com/box/sdk/BoxUser.java

```diff
@@ -160,12 +160,60 @@ public static BoxUser getCurrentUser(BoxAPIConnection api) {
      */
     public static Iterable<BoxUser.Info> getAllEnterpriseUsers(final BoxAPIConnection api, final String filterTerm,
             final String... fields) {
+        return getUsersInfoForType(api, filterTerm, null, fields);
+    }
+
+    /**
+     * Gets a limited set of information about an external user.  (A user collaborating
+     * on content owned by the enterprise).
+     * @param  api        the API connection to be used when retrieving the users.
+     * @param  filterTerm used to filter the results to only users matching the given login.
+     *                    This does exact match only, so if no filter term is passed in, nothing
+     *                    will be returned.
+     * @param  fields     the fields to retrieve. Leave this out for the standard fields.
+     * @return an iterable containing external users matching the given email
+     */
+    public static Iterable<BoxUser.Info> getExternalUsers(final BoxAPIConnection api, final String filterTerm,
+          final String... fields) {
+        return getUsersInfoForType(api, filterTerm, "external", fields);
+    }
+
+    /**
+     * Gets any managed users that match the filter term as well as any external users that
+     * match the filter term.  For managed users it matches any users names or emails that
+     * start with the term.  For external, it only does full match on email.  This method
+     * is ideal to use in the case where you have a full email for a user and you don't
+     * know if they're managed or external.
+     * @param  api        the API connection to be used when retrieving the users.
+     * @param filterTerm    The filter term to lookup users by (login for external, login or name for managed)
+     * @param fields        the fields to retrieve. Leave this out for the standard fields.
+     * @return an iterable containing users matching the given email
+     */
+    public static Iterable<BoxUser.Info> getManagedOrExternalUsers(final BoxAPIConnection api,
```

Comment:
What do you think of calling this `getEnterpriseAndExternalUsers`? I'm wondering if that would be more consistent with the other `getAllEnterpriseUsers` method.